### DISCUSSION TITLE
Xray, radio and optical/IR in CIGALE v2022.0 and binning of bayes results

### DISCUSSION THREAD

**Discussion by olliepye** [mboquien/cigale#9] [Q&A] [answered]
Hi All,

I have a few questions in regards to the use of CIGALE v2022.0. 

Firstly, is it possible to use X-ray, radio and optical/IR data in a single analysis? I note that in Yang+22, X-ray fluxes are not included in the radio analysis to keep SED fits and source classification independent of X-ray information. I am using the ZFOURGE catalogue which contains optical, xray and radio bands. Will this cause degeneracies? 

My second question is in regards to stellar mass and star formation rate estimation. When plotting the best fit results for stellar mass and SFR, the data is binned as shown here:

![best_sfr_mass](https://user-images.githubusercontent.com/42687811/166857365-7478f0d5-b2fa-4a93-b1af-5b2f6bd86005.png)

This is expected due to the discrete parameter space. However, when plotting the bayes results, I would expect this binning to be removed since we are taking the weighted sum of the models. The bayes results give:

![bayes_sfr_mass](https://user-images.githubusercontent.com/42687811/166857360-c668b4ad-0ca7-4118-b934-2b057a63251b.png)

There is some difference, however there is still significant binning. I was wondering if this behaviour was expected? Or does this indicate a problem with the selected parameter space? For reference, I have also attached the mock mass and sfr plots. 

Thanks, Ollie 

|![mockmass_crop](https://user-images.githubusercontent.com/42687811/166858333-b43bba04-9278-4b89-aaf0-6ded24ee0933.png)|![mocksfr_crop](https://user-images.githubusercontent.com/42687811/166858337-bfafb371-9a15-4ff0-8db4-6a2fc6b2ab27.png)|
|---|---|

**Comment by mboquien** [comment, marked as answer]
> Firstly, is it possible to use X-ray, radio and optical/IR data in a single analysis?

It is possible. There is no limitation on this.

>  Or does this indicate a problem with the selected parameter space?

This is probably because the SFH models are too sparse and and in some cases all the Bayesian weight ends up on a single SFH, the others having a much lower likelihood. Something else that could help is to have a somewhat more complex SFH, for instance by having a recent burst with a variable strength.

**Reply by Guang91** [reply]
1. Sure, and I have tested that. Below is an example fit. 
![image](https://user-images.githubusercontent.com/7922328/167037401-7b7004e1-094b-45d3-82ac-cb599038e4fb.png)

2. I agree with @mboquien. A more dense/complex SFH could make the plot more smooth.

**Reply by olliepye** [reply]
Thank you both @mboquien and @Guang91 for your responses!
